packages/components/src/internal/components/editable/models.ts

```diff
@@ -512,6 +543,20 @@ export class EditorModel
         return List<ValueDescriptor>();
     }
 
+    isReadOnlyRow(rowIdx: number, readonlyRows: string[]): boolean {
+        const pkValue = this.getPkValue(rowIdx).toString();
```

Comment:
Add defensive check for `this.getPkValue(rowIdx)?.toString()`. Also add `if (pkValue === undefined) return false;`.